app/Http/Controllers/SupplyController.php

```diff
@@ -0,0 +1,269 @@
+        $em->persist($log);
+        $em->flush();
+
+        $url = Utils::addUrlParameter($url, 'pid', $log->getId());
+
+        $adselectService = $this->container->has('adselect') ? $this->container->get('adselect') : null;
+        $adselectService instanceof Adselect;
+
+        if ($adselectService) {
+            $adselectService->addImpressions([
+                $log->getAdselectJson(),
+            ]);
+        }
+
+        $response = new RedirectResponse($url);
+        return $response;
+    }
+
+    public function logNetworkView(Request $request, $id)
+    {
+        if ($request->query->get('r')) {
+            $url = Utils::UrlSafeBase64Decode($request->query->get('r'));
+            $request->query->remove('r');
+        }
+
+        $qString = http_build_query($request->query->all());
+        if ($qString) {
+            $qPos = strpos($url, '?');
+
+            if ($qPos === false) {
+                $url .= '?' . $qString;
+            } elseif ($qPos == strlen($url) - 1) {
+                $url .= $qString;
+            } else {
+                $url .= '&' . $qString;
+            }
+        }
+
+        $logIp = bin2hex(inet_pton($request->getClientIp()));
+
+        $cid = Utils::getRawTrackingId($request->query->get('cid'));
+        $tid = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
+        $payFrom = $request->query->get('pfr');
+
+        $log = new NetworkEventLog();
+        $log->cid = $cid;
+        $log->banner_id = $id;
+        $log->pay_from = $payFrom;
+        $log->tid = $tid;
+        $log->ip = $logIp;
+        $log->event_type = "view";
+        $log->context = Utils::getImpressionContext($request);
+
+        // GET kewords from aduser
+        $impressionId = $request->query->get('iid');
+        $aduser_endpoint = config('app.aduser_endpoint');
+        if ($aduser_endpoint && $impressionId) {
+            $userdata = json_decode(file_get_contents("{$aduser_endpoint}/get/{$impressionId}"), true);
+        } else {
+            $userdata = [];
+        }
+        $log->our_userdata = $userdata['keywords'];
+        $log->human_score = $userdata['human_score'];
+        $log->user_id = $userdata['user_id'];
+        $log->save();
+
+        // $adselectService = $this->container->has('adselect') ? $this->container->get('adselect') : null;
+        // $adselectService instanceof Adselect;
+        //
+        // if ($adselectService) {
+        //     $adselectService->addImpressions([
+        //         $log->getAdselectJson(),
+        //     ]);
+        // }
+
+        $backUrl = route('log-network-click', ['log_id' => $log->id]);
+
+        $url = Utils::addUrlParameter($url, 'pid', $log->id);
+        $url = Utils::addUrlParameter($url, 'k', Utils::UrlSafeBase64Encode(json_encode($log->our_userdata)));
+        $url = Utils::addUrlParameter($url, 'r', Utils::UrlSafeBase64Encode($backUrl));
+
+        $response = new RedirectResponse($url);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)